cps/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java

```diff
@@ -224,11 +227,12 @@ public String getDisplayName() {
             throw new IOException(e);
         }
     }
-    
+
+    private static final ExecutorService getProgramPromiseExecutorService = Executors.newCachedThreadPool(new NamingThreadFactory(new DaemonThreadFactory(), "CpsStepContext.getProgramPromise"));
```

Comment:
`ContextResettingExecutorService`? I do not think it matters (and `Timer` is not, currently), but I suppose it cannot hurt.